data-prepper-plugins/kafka-source/src/main/java/org/opensearch/dataprepper/plugins/kafka/source/KafkaSource.java

```diff
@@ -0,0 +1,126 @@
+  private final KafkaSourceConfig sourceConfig;
+  private Set<MultithreadedConsumer> consumers = new HashSet<>();
+  private ExecutorService executorService;
+  private static final String KAFKA_WORKER_THREAD_PROCESSING_ERRORS = "kafkaWorkerThreadProcessingErrors";
+  private final Counter kafkaWorkerThreadProcessingErrors;
+  private final PluginMetrics pluginMetrics;
+  private String consumerGroups;
+  private MultithreadedConsumer multithreadedConsumer;
+  private int workers = 0;
+
+  @DataPrepperPluginConstructor
+  public KafkaSource(final KafkaSourceConfig sourceConfig, final PluginMetrics pluginMetrics) {
+    this.sourceConfig = sourceConfig;
+    this.pluginMetrics = pluginMetrics;
+    this.kafkaWorkerThreadProcessingErrors = pluginMetrics.counter(KAFKA_WORKER_THREAD_PROCESSING_ERRORS);
+  }
+  
+  @Override
+  public void start(Buffer<Record<Object>> buffer) {
+    sourceConfig.getTopics().forEach(topic -> {
+      Properties consumerProperties = getConsumerProperties(topic);
+        try {
+        	workers = workers + topic.getTopic().getConsumerGroupConfig().getWorkers(); 
+        	consumerGroups = topic.getTopic().getConsumerGroupConfig().getGroupName();
+            IntStream.range(0, workers).forEach(index -> {
+              String consumerId = Integer.toString(index + 1);
+              multithreadedConsumer = new MultithreadedConsumer(consumerId,
+                      consumerGroups, consumerProperties, topic.getTopic(), buffer, pluginMetrics);
+              consumers.add(multithreadedConsumer);
+            });
+        } catch (Exception e) {
+          LOG.error("Failed to setup the Kafka Source Plugin.", e);
+          kafkaWorkerThreadProcessingErrors.increment();
+        }
+    });
+      executorService = Executors.newFixedThreadPool(workers);
+      try {
+         executorService.invokeAll(consumers);
+      } catch (InterruptedException e) {
+        LOG.error("Kafka source executor service is failed to invoke thread-pool...");
+        Thread.currentThread().interrupt();
+      }
+  }
+
+  @Override
+  public void stop() {
+   LOG.info("Shutting down Consumers...");
+   executorService.shutdown();
+   try {
+      if (!executorService.awaitTermination(
+          sourceConfig.getTopics().get(0).getTopic().getConsumerGroupConfig().getThreadWaitingTime().toSeconds(), TimeUnit.MILLISECONDS)) {
+    	  LOG.info("Consumer threads are waiting for shutting down...");
+        executorService.shutdownNow();
+      }
+    } catch (InterruptedException e) {
+      if (e.getCause() instanceof InterruptedException) {
+        LOG.error("Interrupted during consumer shutdown, exiting uncleanly...");
+        executorService.shutdownNow();
+        Thread.currentThread().interrupt();
+      }
+    }
+    LOG.info("Consumer shutdown successfully...");
+  }
+
+  private Properties getConsumerProperties(TopicsConfig topicConfig) {
+    Properties properties = new Properties();
+    try {
```

Comment:
I dont see SECURITY_PROTOCOL_CONFIG here so this defaults to PLAINTEXT. are we going to add sasl etc config later?